### ISSUE TITLE
Local Exploit Suggest Module - Failed bash not found

### ISSUE THREAD

**Issue by 0xskunk** [rapid7/metasploit-framework#13850]
## Steps to reproduce

How'd you do it?
```
1. Background a reverse shell session I have on a target with Ctrl-Z.
2. use post/multi/recon/local_exploit_suggester
3. set session X
4. Run
```

## Expected behavior

Local exploits usually load and suggest potential privilege escalation vectors.

## Current behavior

```
msf5 post(multi/recon/local_exploit_suggester) > run

[*] 10.13.1.160 - Collecting local exploits for x86/linux...
[*] 10.13.1.160 - 35 exploit checks are being tried...
[-] 10.13.1.160 - Post failed: Msf::Exploit::Failed bash not found
[-] 10.13.1.160 - Call stack:
[-] 10.13.1.160 -   /usr/share/metasploit-framework/lib/msf/core/exploit.rb:1445:in `fail_with'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb:178:in `check_for_bash'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb:204:in `check'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:121:in `block in run'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:119:in `each'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:119:in `run'
[*] Post module execution completed
msf5 post(multi/recon/local_exploit_suggester) > 
```
## System stuff
Debian (Kali) 5.6.14

### Metasploit version
```
Framework: 5.0.97-dev
Console  : 5.0.97-dev
```

### I installed Metasploit with:
Kali was installed upon setting up this VM.

### OS

Linux - Debian

**Comment by h00die** [comment]
can you try `exploits/linux/local/exim4_deliver_message_priv_esc` against that session?  I think the `check` command should be enough.  Fairly certain it will raise the same error, just want to verify.

Also, what is the OS and version of the target?

**Comment by bcoles** [comment]
This is due to using `fail_with` inside a module `check` method.

The following patch should resolve the issue.

```diff
diff --git a/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb b/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb
index 4fd39f61f8..7ef91105d2 100644
--- a/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb
+++ b/modules/exploits/linux/local/exim4_deliver_message_priv_esc.rb
@@ -201,7 +201,9 @@ class MetasploitModule < Msf::Exploit::Local
       socket.close
       socket_subsystem.shutdown
     else
-      check_for_bash
+      if session.type == 'shell' && !command_exists?('/bin/bash')
+        return CheckCode::Safe('bash not found')
+      end
       res = cmd_exec("/bin/bash -c 'exec 3</dev/tcp/localhost/#{datastore['EXIMPORT']} && "\
                      "(read -u 3 && echo $REPLY) || echo false'")
       if res == 'false'
```

**Comment by 0xskunk** [comment]
Hi, thanks for the response. I've made the changes suggested by @bcoles, and just tried to run it again and the check_for_bash issue is gone. However, it's now throwing a different set of errors. I can open a new issue if needed. I've upgraded and updated my metasploit and restarted my system.

```
[-] 10.13.1.160 - Post failed: RuntimeError Could not determine glibc version
[-] 10.13.1.160 - Call stack:
[-] 10.13.1.160 -   /usr/share/metasploit-framework/lib/msf/core/post/linux/system.rb:313:in `rescue in glibc_version'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/lib/msf/core/post/linux/system.rb:309:in `glibc_version'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/exploits/linux/local/glibc_origin_expansion_priv_esc.rb:94:in `check'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:121:in `block in run'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:119:in `each'
[-] 10.13.1.160 -   /usr/share/metasploit-framework/modules/post/multi/recon/local_exploit_suggester.rb:119:in `run'
[*] Post module execution completed
```

@h00die Target details are below. It's intended to be an outdated glibc privilege escalation box. I have been able to exploit it using the ld_audit module, but still keen to get to the bottom of my issues with the local exploit suggester.

```
uname -a 
Linux core 2.6.32-21-generic-pae #32-Ubuntu SMP Fri Apr 16 09:39:35 UTC 2010 i686 GNU/Linux
ldd --version
ldd (Ubuntu EGLIBC 2.11.1-0ubuntu7) 2.11.1
```

**Comment by bcoles** [comment]
> Hi, thanks for the response. I've made the changes suggested by @bcoles, and just tried to run it again and the check_for_bash issue is gone.

There's also a PR up to fix this issue #13859.

> However, it's now throwing a different set of errors. I can open a new issue if needed. I've upgraded and updated my metasploit and restarted my system.

No need to restart your system between changes. Yes, this is a different error / issue. Please create a new issue.

**Comment by 0xskunk** [comment]
Since the bash issue is fixed, I'll close this report now. Thanks for your help gents.
